packages/kit/types/helper.d.ts

```diff
@@ -26,7 +26,8 @@ export type Location<Params extends Record<string, string> = Record<string, stri
 	query: URLSearchParams;
 };
 
-export type MaybePromise<T> = T | Promise<T>;
+export type Index<T = any> = Record<string, T>;
```

Comment:
I'm not totally in love with the name `Index` though having a slightly hard time coming up with a better name. I think `Indexed` sounds a bit better but might be confused with something like [indexed access types](https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html). I sort of wonder if it's even worth adding `Index`. I do like that it's shorter, but on the other hand I think it's less clear what it is and it adds a layer of indirection in reading the code

Reply:
That's what `Record<K, T>` is for, and I think it's worth it, especially when used multiple times like in `page.d.ts`. It does abstracts the code a bit, but anyone reading would only need to see it once to know what it means.

We can rename to make it clearer too. In that case, maybe `Indexable`, `Indexed`, or `IndexSig`?

Reply:
Maybe `Rec`? It makes it a bit clearer that it's related to a record. Things with index sound a bit wrong to me also because you can create, for example, a database index on non-string fields. So if index doesn't imply `string`, why use a term unrelated to `Record` in that case?

Reply:
Sure, it's just a helper type after all